conans/client/conf/config_installer.py

```diff
@@ -1,12 +1,34 @@
 import os
-from conans.tools import unzip
 import shutil
+from six.moves.urllib.parse import urlparse
+
+from conans.tools import unzip
 from conans.util.files import rmdir, mkdir
 from conans.client.remote_registry import RemoteRegistry
 from conans import tools
 from conans.errors import ConanException
 
 
+def _remove_credentials(resource):
+    """
+    Remove credentials from resource url/file path
+
+    :param resource: string with url or file path
+    :return: resource with credentials removed if present
+    """
+    if not resource:
+        return resource
+
+    parsed_url = urlparse(resource)
+    if parsed_url.password is not None:
+        # Remove from netloc credentials, format <username>:<password>@<host>:<port>
+        # see https://tools.ietf.org/html/rfc1738#section-3.1
+        new_netloc = ''.join(parsed_url.netloc.rsplit("@", 1)[1:])
+        # noinspection PyProtectedMember
+        return parsed_url._replace(netloc=new_netloc).geturl()
```

Comment:
Protected methods are protected for reasons. Don't access them, specially in third party library code, unless it is strictly necessary. 
There are many ways to not need this ``._replace()`` here:
- Replace it in the original string
- Compose a new url with the ``parsed_url`` fields.

Reply:
@memsharded  ParseResult is an instance of namedtuple.

_replace is a part of namedtuple public API. It just starts with the underscore to avoid conflicts with the field names. See https://docs.python.org/2/library/collections.html#namedtuple-factory-function-for-tuples-with-named-fields

See more details https://stackoverflow.com/questions/21628852/changing-hostname-in-a-url

Reply:
I see, but I still think this is a symptom of this method being unnecessarily complex. You are calling ``.geturl()`` and ``_replace()`` on the same line, which suggest this is an specialized object (type ``urlparse.ParseResult``), and thus ``_replace()`` reads like a protected method, not the one from a ``namedtuple`` which happens to be the base class of  ``urlparse.ParseResult``.

I might be missing something, but I would suggest keeping it as simple as possible, something like:

```python
def _remove_credentials(resource):
    password = urlparse(resource).password
    return resource.replace(password, "") if password else resource
```

Reply:
@memsharded your code is keeping username which is part of the credentials and also keeps ``@``,

For ```https://test_username:test_password_123@server.com/resource.zip```
returns ```https://test_username:@server.com/resource.zip```

The proposed code returns ```https://server.com/resource.zip``` which reads better in the output ...

If you think that your proposal is the right thing to do please let me know ...

Reply:
This could work:

```
def _remove_credentials(resource):
    parsed = urlparse(resource)
    if not parsed.password:
        return resource
    credentials = parsed.netloc.split("@")[0]
    return resource.replace(credentials, "xxxx:xxxx")
```

Reply:
Ok, I've updated the code with the suggested changes from @memsharded. Check the latest version, hope this time it will meet the requirements.